nextflow-training/exercise2/solution2/nextflow.config

```diff
@@ -0,0 +1,23 @@
+// When a pipeline script is launched, Nextflow looks for this nextflow.config file.
+// This file defines settings and properties and can be organized into different scopes.
+
+// The manifest scope lets you define some metadata.
+// The settings available are author, description, defaultBranch, homePage, mainScript, name, nextflowVersion, and version
+manifest {
+    author = 'Natalie Perez'
+    description = 'This workflow will print a hello message and store it to a file named Hello.txt'
+}
+
+// The params scope lets you define parameters that will be accessible in the pipeline script.
+params {
+    input_sam = '/root/bcc2020-training/data/mini.sam'
+    flagstat_output_name = 'NA12878_chr20_ds.sam.flagstat.metrics'
```

Comment:
output name should be mini.sam.metrics, is there a way to parameterize this name dynamically based on input name?

Reply:
oh wait I see that this file is like the parameter JSON equivalent right?

Reply:
Yeah, I've been thinking of it as somewhat equivalent to the parameter json file

Reply:
okay, well solution 2 is supposed to be about parameterizing outputs, ideally dynamically.
So the simple route is how you have it except the name is wrong, should be mini.sam.metrics.
The ideal route would be dynamically constructing flagstat_output_name from input sam name, kinda how you have the container coming from params.docker_image, if thats possible.

Reply:
not quite how you had it, but closer I think